src/utils/COMETAdapter.cpp

```diff
@@ -0,0 +1,577 @@
+*/
+
+// We do not want this class to show up in the docu:
+/// @cond TOPPCLASSES
+
+
+class TOPPCOMETAdapter :
+  public TOPPBase
+{
+public:
+  TOPPCOMETAdapter() :
+    TOPPBase("COMETAdapter", "Annotates MS/MS spectra using COMET.")
+  {
+  }
+
+protected:
+  void registerOptionsAndFlags_()
+  {
+
+    registerInputFile_("in", "<file>", "", "Input file");
+    setValidFormats_("in", ListUtils::create<String>("mzML"));
+    registerOutputFile_("out", "<file>", "", "Output file");
+    setValidFormats_("out", ListUtils::create<String>("idXML"));
+    registerInputFile_("database", "<file>", "", "FASTA file or pro file. Non-existing relative file-names are looked up via'OpenMS.ini:id_db_dir'", true, false, ListUtils::create<String>("skipexists"));
+    setValidFormats_("database", ListUtils::create<String>("FASTA"));
+    registerInputFile_("comet_executable", "<executable>",
+      // choose the default value according to the platform where it will be executed
+      "comet.exe",                     
+      "Comet executable of the installation e.g. 'comet.exe'", true, false, ListUtils::create<String>("skipexists"));
+
+    //
+    // Optional parameters //
+    //
+
+    //Files
+    registerOutputFile_("pin_out", "<file>", "", "Output file - for percolator input", false);
+    setValidFormats_("pin_out", ListUtils::create<String>("pin"));
+    registerInputFile_("default_params_file", "<file>", "", "Default Comet params file. All parameters of this take precedence. A template file can be generated using comet.exe -p", false, false, ListUtils::create<String>("skipexists"));
+    setValidFormats_("default_params_file", ListUtils::create<String>("txt"));
+    registerIntOption_("threads", "<num>", 1, "number of threads", false, true);
+
+    //Masses
+    registerDoubleOption_("precursor_mass_tolerance", "<tolerance>", 10.0, "precursor_mass_tolerance (MSGF+), peptide_mass_tolerance(COMET)", false, false);
+    // TO DO: change selection menue from numbers to valid strings
+    //registerStringOption_("precursor_error_units", "<unit>", "ppm", "Parent monoisotopic mass error units", true);
+    //vector<String> valid_strings = ListUtils::create<String>("ppm,Da");
+    registerIntOption_("precursor_error_units", "<search_enzyme_number>", 2, "peptide_mass_units (COMET) 0=amu, 1=mmu, 2=ppm", false, false);
+    registerIntOption_("mass_type_parent", "<num>", 1, "0=average masses, 1=monoisotopic masses", false, true);
+    registerIntOption_("mass_type_fragment", "<num>", 1, "0=average masses, 1=monoisotopic masses", false, true);
+    registerIntOption_("precursor_tolerance_type", "<num>", 0, "0=average masses, 1=monoisotopic masses", false, false);
+    registerIntOption_("isotope_error", "<num>", 0, "0=off, 1=on -1/0/1/2/3 (standard C13 error), 2= -8/-4/0/4/8 (for +4/+8 labeling)", false, false);
+
+    //Search Enzyme
+    vector<String> all_enzymes;
+    EnzymesDB::getInstance()->getAllXTandemNames(all_enzymes);
```

Comment:
This is a function in EnzymeDB that goes through all the entries, looks for the XTandem names of them (if present) and adds them to the String vector.

You have three options (sorted ascending by effort but descending by beauty):
- Hardcode the names to the ones out-of-the-box supported by Comet (change your map from name -> number accordingly)
- Add Comet numbers to share/CHEMISTRY/Enzymes.xml, write an EnzymeDB::getAllCometNames and an Enzyme::getCometID (replacing your map); see XTandemAdapter for this
- Use getAllNames (so the user can potentially use all of ours) and write a function to add this Enzyme to the end of the Comet file in the enzyme section (potentially with some helper functions in the Enzyme class)